helix-core/src/text_annotations.rs

```diff
@@ -0,0 +1,271 @@
+use std::cell::Cell;
+use std::convert::identity;
+use std::ops::Range;
+use std::rc::Rc;
+
+use crate::syntax::Highlight;
+use crate::Tendril;
+
+/// An inline annotation is continuos text show
+/// on the screen before the grapheme that starts at
+/// `char_idx`
+#[derive(Debug, Clone)]
+pub struct InlineAnnotation {
+    pub text: Tendril,
+    pub char_idx: usize,
+}
+
+/// Represents a *single Grapheme** that is part of the document
```

Comment:
```suggestion
/// Represents a **single Grapheme** that is part of the document
```

I'm not sure if rust markdown is single or double `*` for bold though